BootloaderCorePkg/Library/S3SaveRestoreLib/S3SaveRestore.c

```diff
@@ -95,8 +95,18 @@ AppendS3Info (
   if ( (UINT8 *)CommHdr + ((BL_PLD_COMM_HDR*)DataPtr)->TotalSize > SmmBase + SmmSize ) {
     return EFI_OUT_OF_RESOURCES;
   }
+
   CopyMem ((VOID *)CommHdr, DataPtr, ((BL_PLD_COMM_HDR*)DataPtr)->TotalSize);
 
+  //
+  // If the ID=SMMBASE_INFO_COMM_ID, then we are only copying the
+  // header and rest of TotalSize is placeholder for payload to fill in.
+  // So, do a ZeroMem on SmmBaseInfo data (TotalSize - sizeof(header)) 
+  //
+  if (CommHdr->Id == SMMBASE_INFO_COMM_ID) {

```

Comment:
this seems to be very specific to certain ID.
Do you think we should add a function parameter to indicate we want to zero the payload part or not while appending ?

Reply:
Agree. Will add another param to indicate valid payload data.